drivers/serial/uart_mcux_iuart.c

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2019, Manivannan Sadhasivam <mani@kernel.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_kinetis_iuart
+
+#include <device.h>
+#include <drivers/uart.h>
+#include <drivers/clock_control.h>
+#include <errno.h>
+#include <fsl_uart.h>
+#include <soc.h>
+
+struct mcux_iuart_config {
+	UART_Type *base;
+	char *clock_name;
```

Comment:
This can be made `const` assuming that this will never change during the program life cycle.. 

Making things `const` makes them go to the ROM which helps saving RAM (in MCUs, we generally have more ROM than RAM)..